### PR TITLE
make normalize ticksPerWheel more stable and high-performance

### PR DESCRIPTION
**Pull request by mmaxiaolei**
make normalize ticksPerWheel more stable and high-performance

### CODE REVIEW

File: common/src/main/java/io/netty/util/HashedWheelTimer.java [hunk 1]
```diff
@@ -294,11 +294,13 @@ public HashedWheelTimer(
...
-            normalizedTicksPerWheel <<= 1;
-        }
-        return normalizedTicksPerWheel;
+        int n = ticksPerWheel - 1;
+        n |= n >>> 1;
+        n |= n >>> 2;
+        n |= n >>> 4;
+        n |= n >>> 8;
+        n |= n >>> 16;
+        return (n < 0) ? 1 : (n >= 1073741824) ? 1073741824 : n + 1;
```
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/HashedWheelTimer.java]: Can you make `1073741824` an static constant with a descriptive name ?

Comment (normanmaurer) [inline common/src/main/java/io/netty/util/HashedWheelTimer.java]: Also share it with other places in the code.

**Comment by normanmaurer** [comment]
@mmaxiaolei also please sign our icla and adjust commit message to match our template:

http://netty.io/s/ilcla
http://netty.io/wiki/writing-a-commit-message.html

**Comment by slandelle** [comment]
> high-performance

@mmaxiaolei Could you please share some numbers?

**Comment by normanmaurer** [comment]
@mmaxiaolei @slandelle also I wonder if it really makes a lot of difference as this one is only called in the constructor anyway.

**Comment by Scottmitch** [comment]
I agree with @slandelle and @normanmaurer ... please demonstrate the value this provides via JMH benchmark. Thanks.

**Comment by fenik17** [comment]
IMO benchmark not necessary. This optimization used also in JDK, referring to book "Hacker's Delight" sec 3-2 "Rounding Up/Down to the Next Power of 2".
For example look at ConcurrentHashMap.tableSizeFor().

File: common/src/main/java/io/netty/util/HashedWheelTimer.java (hunk 1 above)
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/HashedWheelTimer.java]: @mmaxiaolei could you please address my comment ?

**Comment by normanmaurer** [comment]
I am closing this now as it seems like @mmaxiaolei is not interested in this anymore. Please reopen if still interested and address the comments

**Pull request closed without merging**